flink-table/flink-sql-gateway/src/main/java/org/apache/flink/table/gateway/service/context/SessionContext.java

```diff
@@ -269,6 +238,48 @@ public static SessionContext create(
     // Helpers
     // ------------------------------------------------------------------------------------------------------------------
 
+    private static CatalogManager buildCatalogManager(
+            Configuration configuration,
+            URLClassLoader userClassLoader,
+            SessionEnvironment environment) {
+        CatalogManager.Builder builder =
+                CatalogManager.newBuilder()
+                        // Currently, the classloader is only used by DataTypeFactory.
+                        .classLoader(userClassLoader)
+                        .config(configuration);
+
+        // init default catalog
+        String defaultCatalogName;
+        Catalog defaultCatalog;
+        if (environment.getDefaultCatalog().isPresent()) {
+            defaultCatalogName = environment.getDefaultCatalog().get();
+            defaultCatalog = environment.getRegisteredCatalogs().get(defaultCatalogName);
+        } else {
+            EnvironmentSettings settings =
+                    EnvironmentSettings.newInstance().withConfiguration(configuration).build();
+            defaultCatalogName = settings.getBuiltInCatalogName();
```

Comment:
It's better if we can check whether the defaultCatalogName also exists in the `SessionEnvironment`.

```
         if (environment.getRegisteredCatalogs().containsKey(defaultCatalogName)) {
                throw new SqlGatewayException(
                        String.format(
                                "The name of the registered catalog is conflicts with the built-in default catalog name: %s.",
                                defaultCatalogName));
            }
```